unstructured/documents/elements.py

```diff
@@ -667,8 +677,12 @@ def __init__(
         # -- defined in a subclass.
         self.text = self.text if hasattr(self, "text") else ""
 
+    @property
+    def id(self) -> str | uuid.UUID | NoID | UUID:
+        return self._id
```

Comment:
`Element` has a very basic behavior, concrete implementation that overrides it is in the `Text` class

Reply:
`Element` is a base-class (only) as far as I know and is never instantiated directly, only one of its subclasses (at least in a partitioning element-stream). So I'd be inclined to either remove this property or make its return type `str` (consistent with subtypes) and unconditionally raise `AttributeError` with something like `"an instance of the Element base-class cannot have an id"`